src/components/connect.js

```diff
@@ -47,14 +47,15 @@ export default function connect(mapPropsToRequestsToProps, options = {}) {
     }
 
     invariant(isPlainObject(mapping), 'Request for `%s` must be either a string or a plain object. Instead received %s', prop, mapping)
-    invariant(mapping.url, 'Request object for `%s` must have `url` attribute.', prop)
+    invariant(mapping.url || mapping.value, 'Request object for `%s` must have `url` (or `value`) attribute.', prop)
+    invariant(!(mapping.url && mapping.value), 'Request object for `%s` must not have both `url` and `value` attributes.', prop)
 
     mapping.equals = function (that) {
       if (this.comparison !== undefined) {
         return this.comparison === that.comparison
       }
 
-      return [ 'url', 'method', 'headers', 'body' ].every((c) => {
+      return [ 'value', 'url', 'method', 'headers', 'body' ].every((c) => {
```

Comment:
@georgebonnr Correct, it will not prevent a function from being called to create `value`, but it will prevent the `PromiseState` from being updated with the same `value`. If you want to memoize the function result, it would need to happen outside of `connect()` or possibly use `comparison` to have finer grain control of when the `PromiseState` is updated, but not sure it would really help much for something just based on a top-level prop. As you mentioned, the typical case inside of `then` or `andThen` will stop the function from being called unnecessarily.